src/main/resources/db/migration/common/V2024.11.09.0002__add_note_table.sql

```diff
@@ -0,0 +1,26 @@
+-- table note
+CREATE TABLE note
+(
+    id                 UUID         PRIMARY KEY,
+    reference          UUID         NOT NULL,
+    content            TEXT         NOT NULL,
+    prison_number      VARCHAR(10)  NOT NULL,
+    created_at         TIMESTAMP(3) WITH TIME ZONE NOT NULL,
+    created_at_prison  VARCHAR(10)  NOT NULL,
+    created_by         VARCHAR(50)  NOT NULL,
+    updated_at         TIMESTAMP(3) WITH TIME ZONE NOT NULL,
+    updated_by         VARCHAR(50)  NOT NULL,
+    updated_at_prison  VARCHAR(10)  NOT NULL,
+    note_type          VARCHAR(10)  NOT NULL,
+    entity_type        VARCHAR(50)  NOT NULL,
+    entity_reference   UUID         NOT NULL
+);
+
+CREATE UNIQUE INDEX note_reference_idx ON note
+(
+    reference
+);
+CREATE INDEX prison_number_idx ON note
+(
+    prison_number
+);
```

Comment:
We need some sql to migrate the existing notes fields out of existing goals into this new structure, and then drop the notes column from the goal table.
Your SQL will need to generate UUIDs for the ID and reference columns, content, prison_number, entity_reference, createdAt/updatedAt etc fields should just be able to copy from the Goal record. And note type and entity type can be hardcoded because we know what they are (because we only support one type at the moment)